From 3c0413bc653de9994e7b09c7d119d3cea88ce45b Mon Sep 17 00:00:00 2001
From: Michel JAOUEN <michel.jaouen@st.com>
Date: Wed, 17 Jul 2024 15:50:05 +0200
Subject: [PATCH 683/954] fix(stm32mp2): remove all set of gpio secure in M33TD
 build

In m33td flavor, tf-a cannot set the secure attribute, it is done
by m33.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git dec3a80687a24cf99bd09e8e0e71c1fe6b1e3877 fix(stm32mp2): remove all set of gpio secure in M33TD build]
Change-Id: I86679cbb2be8e4c9dc652e9de9537bb3153d3772
Signed-off-by: Michel JAOUEN <michel.jaouen@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/403152
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/st/gpio/stm32_gpio.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/st/gpio/stm32_gpio.c b/drivers/st/gpio/stm32_gpio.c
index a068778461..c3d5b26f4d 100644
--- a/drivers/st/gpio/stm32_gpio.c
+++ b/drivers/st/gpio/stm32_gpio.c
@@ -298,7 +298,9 @@ static void set_gpio(uint32_t bank, uint32_t pin, uint32_t mode, uint32_t type,
 #endif
 	}
 #else /* !STM32MP13 && !STM32MP15 */
+#if !STM32MP_M33_TDCID
 	set_gpio_secure_cfg(bank, pin, true);
+#endif
 #endif /* STM32MP13 || STM32MP15 */
 }
 
-- 
2.53.0

